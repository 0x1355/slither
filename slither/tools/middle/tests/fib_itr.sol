pragma solidity ^0.6.4;

contract Fib {
    function fib(uint n) public returns(uint res) {
        uint prev_prev_num;
        uint prev_num = 0;
        uint num = 1;
        for (uint i = 1; i < n; i++) {
            prev_prev_num = prev_num;
            prev_num = num;
            num = prev_prev_num + prev_num;
        }
        return num;
    }
}