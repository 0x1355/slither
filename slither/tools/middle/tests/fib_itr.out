OVERLAY_FUNCTION_
    ENTRY_POINT
    EXPRESSION
    EXPRESSION
    EXPRESSION
    EXPRESSION
    IF_LOOP
    OVERLAY
fib
    ENTRY_POINT
    NEW VARIABLE
    NEW VARIABLE
    NEW VARIABLE
    NEW VARIABLE
    BEGIN_LOOP
    IF_LOOP
    OVERLAY
    END_LOOP
    RETURN
